Prog/vacant-lots-montgomery_YS.Rmd

```diff
@@ -0,0 +1,1257 @@
+5. single-family attached 
+6. single-family semidetached 
+7. single-family detached 
+8. all (dwelling type is NA)
+
+### Create factor variable
+To use this hierarchy, we transform the `dwelling_type` variable into a factor, assign the factors to the appropriate level, (1-6, as listed above).
+We will also create a designation for dwelling types that are restricted to a certain amount of units (one unit for "One Family Dwelling" and one for two units for "Duplex"). 
+
+**NOTE** This will be VERY jurisdiction specific.
+
+**NOTE** If there are no one or two unit limitations, still make the variable and set them all equal to zero. This allows the functionality of the calculation to be more uniform across counties.
+
+```{r dwelling-factor Gaithersburg}
+#YS edits
+
+SF_g <- c("R-A", "R-90", "R-6")
+SFA_g <- c()
+MF_g <- c("RP-T","R-20","R-18","R-H","CB")
+COM_g <- c("CBD","CD","RB")
+MXD_g <- c("MXD")
+
+zoning_gaithersburg2 <- zoning_gaithersburg1 %>% 
+  mutate(dwelling_type=ifelse(is.na(dwelling_type)==1, "all", dwelling_type)) %>% 
+  mutate(dwelling_type = factor(dwelling_type,
+                                levels = c("multiple family condominium dwellings",
+                                           "multiple family dwelling",
+                                           "townhouse",
+                                           "two family dwellings",
+                                           "single-family attached",
+                                           "single-family semidetached",
+                                           "single-family detached",
+                                           "all"
+                                           )),
+         one_unit_flag = ifelse(dwelling_type == "single-family attached" | 
+                                dwelling_type == "single-family semidetached" | 
+                                dwelling_type == "single-family detached",
+                                1,
+                                0),
+         two_unit_flag = ifelse(dwelling_type == "two family dwellings",
+                                1,
+                                0)) %>% 
+        mutate(housing_type= case_when(zoning_code %in% SF_g ~ "SF", 
+                                       zoning_code %in% SFA_g ~ "SFA",
+                                       zoning_code %in% MF_g ~ "MF",
+                                       zoning_code %in% COM_g ~ "Commerical",
+                                       zoning_code %in% MXD_g ~ "MXD")) %>% 
+        mutate(lot_area_min= ifelse(housing_type=="MXD", 1800, lot_area_min),
+               height_ft=ifelse(housing_type=="MXD", 45, height_ft),
+               lot_coverage_max = ifelse(housing_type=="MXD", 0.4, lot_coverage_max),
+               density= ifelse(housing_type=="MXD", 21.5, density)) #assign MXD with R-20
+
+
+```
+
+*For Rockville*
+```{r list codes Rockville}
+zoning_rockville %>% 
+  count(dwelling_type) %>% 
+  knitr::kable()
```

Comment:
The dwelling type information was missing from the zoning spreadsheet we have. But because there's no duplicate codes in the zoning spreadsheet, we don't really need it for creating the factor variable. I'm using the dwelling type information in the zoning ordinance to create categories for the table shells, though.